src/libPersistence/BlockStorage.cpp

```diff
@@ -580,20 +580,10 @@ bool BlockStorage::PutShardStructure(const DequeOfShard& shards,
   return true;
 }
 
-bool BlockStorage::GetShardStructure(DequeOfShard& shards,
-                                     atomic<uint32_t>& myshardId) {
+bool BlockStorage::GetShardStructure(DequeOfShard& shards) {
   LOG_MARKER();
 
-  unsigned int index = 0;
-  string strMyshardId = m_shardStructureDB->Lookup(index++);
-
-  if (strMyshardId.empty()) {
-    LOG_GENERAL(WARNING, "Cannot retrieve sharding structure!");
-    return false;
-  }
-
-  myshardId = stoul(strMyshardId);
-  LOG_GENERAL(INFO, "Retrieved shard ID: " << myshardId);
+  unsigned int index = 1;
```

Comment:
Why use index, just protobuf the entire structure

Reply:
It's a good idea however considering the backward compatible, we should not change current database format but only can make some extension.